Libraries/Lists/VirtualizedList.js

```diff
@@ -675,6 +734,12 @@ class VirtualizedList extends React.PureComponent<Props, State> {
       if (stickyIndicesFromProps.has(ii + stickyOffset)) {
         stickyHeaderIndices.push(cells.length);
       }
+      // remove all components from parentProps so that these components don't cause the CellRenderer to re-render.
+      const parentProps: Props = {...this.props};
+      delete parentProps.ItemSeparatorComponent;
+      delete parentProps.ListEmptyComponent;
+      delete parentProps.ListFooterComponent;
+      delete parentProps.ListHeaderComponent;
```

Comment:
Instead of copying the whole thing and then removing things you don't want to keep, can you explicitly copy just the props that we do want to keep?

Reply:
In terms of perf, seems like copy whole thing and set not used property to null will be much better. https://jsperf.com/delete-vs-undefined-vs-null/16

Reply:
Looking at the original code the only parent props used by CellRenderer are `const {renderItem, getItemLayout} = parentProps;` and then later `!parentProps.debug`.
Couldn't we just pass them as regular props and make CellRenderer a PureComponent? The ShallowCompare used by PureComponent would just work and we don't have the need shallowEqual or deepDiffer. Or perhaps I'm missing something?